### ISSUE TITLE
DNS state is not restored correctly after disconnecting from two VPNs

### ISSUE THREAD

**Issue by jessebye** [Tunnelblick/Tunnelblick#573]
**Describe the bug**
After connecting to two VPNs (both of which have "Set DNS/WINS" set to "Set nameserver"), then disconnecting, DNS state is incorrect and the internet connection of the host is unavailable until it is reset (i.e. via toggling wifi off/on). 

Disconnecting the first VPN shows the following *correct* output:
```
                           13:50:01 *Tunnelblick:  **********************************************
                           13:50:01 *Tunnelblick:  Start of output from client.down.tunnelblick.sh
                           13:50:01 *Tunnelblick:  Cancelled monitoring system configuration changes
                           13:50:01 *Tunnelblick:  Restored State:DNS
                           13:50:01 *Tunnelblick:  Restored Setup:DNS
                           13:50:01 *Tunnelblick:  Restored DNS and SMB settings
                           13:50:01 *Tunnelblick:  Flushed the DNS cache with dscacheutil -flushcache
                           13:50:01 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
                           13:50:01 *Tunnelblick:  End of output from client.down.tunnelblick.sh
                           13:50:01 *Tunnelblick:  **********************************************
```

Disconnecting the second VPN then shows the following *incorrect* output:
```
                           13:52:25 *Tunnelblick:  **********************************************
                           13:52:25 *Tunnelblick:  Start of output from client.down.tunnelblick.sh
                           13:52:26 *Tunnelblick:  WARNING: Not restoring network settings because no saved Tunnelblick DNS information was found.
                           13:52:26 *Tunnelblick:  Flushed the DNS cache with dscacheutil -flushcache
                           13:52:26 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
                           13:52:26 *Tunnelblick:  End of output from client.down.tunnelblick.sh
                           13:52:26 *Tunnelblick:  **********************************************
```

It appears that the "saved Tunnelblick DNS information" is removed by the first disconnect, and the second disconnect is unable to use that saved information to correctly restore the DNS state.

**To Reproduce**
1. Create two VPN configurations, making sure both are set to "Set nameserver" and use the default up/down scripts that are included in Tunnelblick.
2. Connect to both VPNs.
3. Disconnect from one VPN, observe that DNS is still functional.
4. Disconnect from the second VPN. Observe that DNS is left in a bad state and internet access no longer works, even though both VPNs are disconnected.

**Expected behavior**
Tunnelblick should correctly save and restore the DNS state, regardless of the number of VPNs that connect/disconnect.

**Screenshots**
Good output:
![image](https://user-images.githubusercontent.com/8467862/63975846-d0cac200-ca64-11e9-96bb-63f76a8c0955.png)

Bad output:
![image](https://user-images.githubusercontent.com/8467862/63975864-d9bb9380-ca64-11e9-968a-2c66d1c74e83.png)

**Additional context**
n/a

**Diagnostic information**
```
*Tunnelblick: macOS 10.14.6 (18G84); Tunnelblick 3.8.0 (build 5370); prior version 3.7.9a (build 5321); Admin user
git commit c6c984efbc7c7d803728e6299fd40da488bb344e


Configuration PROD

"Sanitized" condensed configuration file for /Users/jbye/Library/Application Support/Tunnelblick/Configurations/PROD.tblk:

client
dev tun
proto udp
remote 3.130.55.186 1194
resolv-retry infinite
nobind
persist-key
persist-tun
remote-cert-tls server
cipher AES-256-CBC
auth SHA256
key-direction 1
verb 3
<ca>
[Security-related line(s) omitted]
</ca>
<cert>
[Security-related line(s) omitted]
</cert>
<key>
[Security-related line(s) omitted]
</key>
auth-user-pass
reneg-sec 0


================================================================================

Files in PROD.tblk:
      Contents/Resources/config.ovpn

================================================================================

Configuration preferences:

-keychainHasUsernameAndPassword = 1
-notOKToCheckThatIPAddressDidNotChangeAfterConnection = 1
-allowChangesToManuallySetNetworkSettings = 0
-alwaysShowLoginWindow = 0
-lastConnectionSucceeded = 1

================================================================================

Wildcard preferences:

-notOKToCheckThatIPAddressDidNotChangeAfterConnection = 1

================================================================================

Program preferences:

skipWarningAboutSimultaneousConnections = 1
skipWarningAboutDnsProblems = 1
launchAtNextLogin = 1
tunnelblickVersionHistory = (
    "3.8.0 (build 5370)",
    "3.7.9a (build 5321)"
)
statusDisplayNumber = 0
lastLaunchTime = 588803081.93196
doNotShowDisconnectedNotificationWindows = 1
lastLanguageAtLaunchWasRTL = 0
connectionWindowDisplayCriteria = showWhenConnecting
maxLogDisplaySize = 102400
lastConnectedDisplayName = PROD
keyboardShortcutIndex = 1
updateCheckAutomatically = 1
NSWindow Frame ConnectingWindow = 765 742 389 187 0 0 1920 1177 
NSWindow Frame SUUpdateAlert = 2090 588 620 392 1440 0 1920 1177 
detailsWindowFrameVersion = 5370
detailsWindowFrame = {{994, 700}, {920, 468}}
detailsWindowLeftFrame = {{0, 0}, {167, 350}}
detailsWindowViewIndex = 0
detailsWindowConfigurationsTabIdentifier = log
leftNavSelectedDisplayName = PROD
AdvancedWindowTabIdentifier = connectingAndDisconnecting
haveDealtWithOldTunTapPreferences = 1
haveDealtWithOldLoginItem = 1
haveDealtWithAfterDisconnect = 1
SUEnableAutomaticChecks = 1
SUScheduledCheckInterval = 86400
SULastCheckTime = 2019-08-29 20:24:42 +0000
SUHasLaunchedBefore = 1
WebKitDefaultFontSize = 16
WebKitStandardFont = Times

================================================================================

Tunnelblick Log:

2019-08-29 13:43:19.481200 *Tunnelblick: macOS 10.14.6 (18G84); Tunnelblick 3.8.0 (build 5370); prior version 3.7.9a (build 5321)
2019-08-29 13:43:19.942329 *Tunnelblick: Attempting connection with PROD using shadow copy; Set nameserver = 769; monitoring connection
2019-08-29 13:43:19.942497 *Tunnelblick: openvpnstart start PROD.tblk 54780 769 0 1 0 1098032 -ptADGNWradsgnw 2.4.7-openssl-1.0.2r
2019-08-29 13:43:20.008177 *Tunnelblick: openvpnstart starting OpenVPN
2019-08-29 13:43:20.179612 OpenVPN 2.4.7 x86_64-apple-darwin [SSL (OpenSSL)] [LZO] [LZ4] [PKCS11] [MH/RECVDA] [AEAD] built on Jul 19 2019
2019-08-29 13:43:20.179687 library versions: OpenSSL 1.0.2r  26 Feb 2019, LZO 2.10
2019-08-29 13:43:20.180767 MANAGEMENT: TCP Socket listening on [AF_INET]127.0.0.1:54780
2019-08-29 13:43:20.180806 Need hold release from management interface, waiting...
2019-08-29 13:43:20.564603 *Tunnelblick: openvpnstart log:
     OpenVPN started successfully.
     Command used to start OpenVPN (one argument per displayed line):
          /Applications/Tunnelblick.app/Contents/Resources/openvpn/openvpn-2.4.7-openssl-1.0.2r/openvpn
          --daemon
          --log /Library/Application Support/Tunnelblick/Logs/-SUsers-Sjbye-SLibrary-SApplication Support-STunnelblick-SConfigurations-SPROD.tblk-SContents-SResources-Sconfig.ovpn.769_0_1_0_1098032.54780.openvpn.log
          --cd /Library/Application Support/Tunnelblick/Users/jbye/PROD.tblk/Contents/Resources
          --machine-readable-output
          --setenv IV_GUI_VER "net.tunnelblick.tunnelblick 5370 3.8.0 (build 5370)"
          --verb 3
          --config /Library/Application Support/Tunnelblick/Users/jbye/PROD.tblk/Contents/Resources/config.ovpn
          --setenv TUNNELBLICK_CONFIG_FOLDER /Library/Application Support/Tunnelblick/Users/jbye/PROD.tblk/Contents/Resources
          --verb 3
          --cd /Library/Application Support/Tunnelblick/Users/jbye/PROD.tblk/Contents/Resources
          --management 127.0.0.1 54780 /Library/Application Support/Tunnelblick/lobglbllpkjmegfngnhddlfmngmekkkmgmedbpfl.mip
          --management-query-passwords
          --management-hold
          --script-security 2
          --route-up /Applications/Tunnelblick.app/Contents/Resources/client.up.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw
          --down /Applications/Tunnelblick.app/Contents/Resources/client.down.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw
2019-08-29 13:43:20.567328 MANAGEMENT: Client connected from [AF_INET]127.0.0.1:54780
2019-08-29 13:43:20.581383 MANAGEMENT: CMD 'pid'
2019-08-29 13:43:20.581462 MANAGEMENT: CMD 'auth-retry interact'
2019-08-29 13:43:20.581502 MANAGEMENT: CMD 'state on'
2019-08-29 13:43:20.581567 MANAGEMENT: CMD 'state'
2019-08-29 13:43:20.581632 MANAGEMENT: CMD 'bytecount 1'
2019-08-29 13:43:20.583233 *Tunnelblick: Established communication with OpenVPN
2019-08-29 13:43:20.583912 *Tunnelblick: >INFO:OpenVPN Management Interface Version 1 -- type 'help' for more info
2019-08-29 13:43:20.584180 MANAGEMENT: CMD 'hold release'
2019-08-29 13:43:20.596050 *Tunnelblick: Obtained VPN username and password from the Keychain
2019-08-29 13:43:20.596304 MANAGEMENT: CMD 'username "Auth" "jbye"'
2019-08-29 13:43:20.596365 MANAGEMENT: CMD 'password [...]'
2019-08-29 13:43:20.596591 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
2019-08-29 13:43:20.597634 TCP/UDP: Preserving recently used remote address: [AF_INET]3.130.55.186:1194
2019-08-29 13:43:20.597706 Socket Buffers: R=[786896->786896] S=[9216->9216]
2019-08-29 13:43:20.597723 UDP link local: (not bound)
2019-08-29 13:43:20.597738 UDP link remote: [AF_INET]3.130.55.186:1194
2019-08-29 13:43:20.597772 MANAGEMENT: >STATE:1567111400,WAIT,,,,,,
2019-08-29 13:43:20.659809 MANAGEMENT: >STATE:1567111400,AUTH,,,,,,
2019-08-29 13:43:20.659854 TLS: Initial packet from [AF_INET]3.130.55.186:1194, sid=e72a8af2 5da27123
2019-08-29 13:43:20.660035 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
2019-08-29 13:43:20.735127 VERIFY OK: depth=1, C=US, ST=CA, L=Auburn, O=Riskalyze, OU=IT, CN=Riskalyze CA, name=server, emailAddress=jbye@riskalyze.com
2019-08-29 13:43:20.735717 VERIFY KU OK
2019-08-29 13:43:20.735741 Validating certificate extended key usage
2019-08-29 13:43:20.735752 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
2019-08-29 13:43:20.735760 VERIFY EKU OK
2019-08-29 13:43:20.735768 VERIFY OK: depth=0, C=US, ST=CA, L=Auburn, O=Riskalyze, OU=IT, CN=server, name=server, emailAddress=jbye@riskalyze.com
2019-08-29 13:43:20.892992 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 ECDHE-RSA-AES256-GCM-SHA384, 4096 bit RSA
2019-08-29 13:43:20.893050 [server] Peer Connection Initiated with [AF_INET]3.130.55.186:1194
2019-08-29 13:43:22.013664 MANAGEMENT: >STATE:1567111402,GET_CONFIG,,,,,,
2019-08-29 13:43:22.013978 SENT CONTROL [server]: 'PUSH_REQUEST' (status=1)
2019-08-29 13:43:27.192670 SENT CONTROL [server]: 'PUSH_REQUEST' (status=1)
2019-08-29 13:43:32.371145 SENT CONTROL [server]: 'PUSH_REQUEST' (status=1)
2019-08-29 13:43:32.435474 PUSH: Received control message: 'PUSH_REPLY,dhcp-option DNS 10.30.0.2,route 10.30.0.0 255.255.0.0,route REDACTED,route REDACTED,route api-REDACTED.duosecurity.com,route-gateway 172.16.2.1,topology subnet,ping 10,ping-restart 120,ifconfig 172.16.2.2 255.255.255.0,peer-id 0,cipher AES-256-GCM'
2019-08-29 13:43:32.435580 Options error: route parameter network/IP 'REDACTED' must be a valid address
2019-08-29 13:43:32.435599 Options error: route parameter network/IP 'REDACTED' must be a valid address
2019-08-29 13:43:32.435610 Options error: route parameter network/IP 'api-REDACTED.duosecurity.com' must be a valid address
2019-08-29 13:43:32.435641 OPTIONS IMPORT: timers and/or timeouts modified
2019-08-29 13:43:32.435651 OPTIONS IMPORT: --ifconfig/up options modified
2019-08-29 13:43:32.435659 OPTIONS IMPORT: route options modified
2019-08-29 13:43:32.435666 OPTIONS IMPORT: route-related options modified
2019-08-29 13:43:32.435673 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
2019-08-29 13:43:32.435680 OPTIONS IMPORT: peer-id set
2019-08-29 13:43:32.435687 OPTIONS IMPORT: adjusting link_mtu to 1624
2019-08-29 13:43:32.435694 OPTIONS IMPORT: data channel crypto options modified
2019-08-29 13:43:32.435728 Data Channel: using negotiated cipher 'AES-256-GCM'
2019-08-29 13:43:32.435801 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2019-08-29 13:43:32.435810 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2019-08-29 13:43:32.436053 Opening utun (connect(AF_SYS_CONTROL)): Resource busy (errno=16)
2019-08-29 13:43:32.436082 Opening utun (connect(AF_SYS_CONTROL)): Resource busy (errno=16)
2019-08-29 13:43:32.436178 Opened utun device utun2
2019-08-29 13:43:32.436222 MANAGEMENT: >STATE:1567111412,ASSIGN_IP,,172.16.2.2,,,,
2019-08-29 13:43:32.436247 /sbin/ifconfig utun2 delete
                           ifconfig: ioctl (SIOCDIFADDR): Can't assign requested address
2019-08-29 13:43:32.439599 NOTE: Tried to delete pre-existing tun/tap instance -- No Problem if failure
2019-08-29 13:43:32.439671 /sbin/ifconfig utun2 172.16.2.2 172.16.2.2 netmask 255.255.255.0 mtu 1500 up
2019-08-29 13:43:32.442530 /sbin/route add -net 172.16.2.0 172.16.2.2 255.255.255.0
                           add net 172.16.2.0: gateway 172.16.2.2
2019-08-29 13:43:32.445690 MANAGEMENT: >STATE:1567111412,ADD_ROUTES,,,,,,
2019-08-29 13:43:32.445717 /sbin/route add -net 10.30.0.0 172.16.2.1 255.255.0.0
                           add net 10.30.0.0: gateway 172.16.2.1
                           13:43:32 *Tunnelblick:  **********************************************
                           13:43:32 *Tunnelblick:  Start of output from client.up.tunnelblick.sh
                           13:43:34 *Tunnelblick:  Retrieved from OpenVPN: name server(s) [ 10.30.0.2 ], search domain(s) [ ] and SMB server(s) [ ] and using default domain name [ openvpn ]
                           13:43:35 *Tunnelblick:  WARNING: Ignoring DomainName 'openvpn' because DomainName was set manually and '-allowChangesToManuallySetNetworkSettings' was not specified
                           13:43:35 *Tunnelblick:  WARNING: Ignoring ServerAddresses '10.30.0.2' because ServerAddresses was set manually and '-allowChangesToManuallySetNetworkSettings' was not specified
                           13:43:35 *Tunnelblick:  Not replacing search domains 'openvpn' with 'openvpn' because the search domains were set manually, '-allowChangesToManuallySetNetworkSettings' was not selected, and 'Prepend domain name to search domains' was not selected
                           13:43:36 *Tunnelblick:  Saved the DNS and SMB configurations so they can be restored
                           13:43:36 *Tunnelblick:  Did not change DNS ServerAddresses setting of '10.20.0.2'
                           13:43:36 *Tunnelblick:  Did not change DNS SearchDomains setting of 'openvpn'
                           13:43:36 *Tunnelblick:  Did not change DNS DomainName setting of 'openvpn'
                           13:43:36 *Tunnelblick:  Did not change SMB NetBIOSName setting of ''
                           13:43:36 *Tunnelblick:  Did not change SMB Workgroup setting of ''
                           13:43:36 *Tunnelblick:  Did not change SMB WINSAddresses setting of ''
                           13:43:36 *Tunnelblick:  DNS servers '10.20.0.2' were set manually
                           13:43:36 *Tunnelblick:  DNS servers '10.20.0.2' will be used for DNS queries when the VPN is active
                           13:43:36 *Tunnelblick:  NOTE: The DNS servers do not include any free public DNS servers known to Tunnelblick. This may cause DNS queries to fail or be intercepted or falsified even if they are directed through the VPN. Specify only known public DNS servers or DNS servers located on the VPN network to avoid such problems.
                           13:43:36 *Tunnelblick:  Flushed the DNS cache via dscacheutil
                           13:43:36 *Tunnelblick:  /usr/sbin/discoveryutil not present. Not flushing the DNS cache via discoveryutil
                           13:43:36 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
                           13:43:36 *Tunnelblick:  Not notifying mDNSResponderHelper that the DNS cache was flushed because it is not running
                           13:43:36 *Tunnelblick:  Setting up to monitor system configuration with process-network-changes
                           /Applications/Tunnelblick.app/Contents/Resources/ProcessNetworkChanges.plist: service already loaded
                           13:43:36 *Tunnelblick:  End of output from client.up.tunnelblick.sh
                           13:43:36 *Tunnelblick:  **********************************************
2019-08-29 13:43:36.861384 Initialization Sequence Completed
2019-08-29 13:43:36.861468 MANAGEMENT: >STATE:1567111416,CONNECTED,SUCCESS,172.16.2.2,3.130.55.186,1194,,
2019-08-29 13:43:37.170148 *Tunnelblick: DNS address 10.20.0.2 is being routed through the VPN
2019-08-29 13:43:37.183017 *Tunnelblick: Warning: DNS server address 127.0.0.1 is not a public IP address and is not being routed through the VPN.


2019-08-29 13:43:46.155678 *Tunnelblick: process-network-changes: A system configuration change was ignored
2019-08-29 13:52:24.727128 *Tunnelblick: Disconnecting; VPN Details… window disconnect button pressed
2019-08-29 13:52:24.872273 *Tunnelblick: Disconnecting using 'kill'
2019-08-29 13:52:25.037028 event_wait : Interrupted system call (code=4)
2019-08-29 13:52:25.039975 /sbin/route delete -net 10.30.0.0 172.16.2.1 255.255.0.0
                           delete net 10.30.0.0: gateway 172.16.2.1
2019-08-29 13:52:25.042551 Closing TUN/TAP interface
2019-08-29 13:52:25.044725 /Applications/Tunnelblick.app/Contents/Resources/client.down.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw utun2 1500 1552 172.16.2.2 255.255.255.0 init
                           13:52:25 *Tunnelblick:  **********************************************
                           13:52:25 *Tunnelblick:  Start of output from client.down.tunnelblick.sh
                           13:52:26 *Tunnelblick:  WARNING: Not restoring network settings because no saved Tunnelblick DNS information was found.
                           13:52:26 *Tunnelblick:  Flushed the DNS cache with dscacheutil -flushcache
                           13:52:26 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
                           13:52:26 *Tunnelblick:  End of output from client.down.tunnelblick.sh
                           13:52:26 *Tunnelblick:  **********************************************
2019-08-29 13:52:26.321052 SIGTERM[hard,] received, process exiting
2019-08-29 13:52:26.321186 MANAGEMENT: >STATE:1567111946,EXITING,SIGTERM,,,,,
2019-08-29 13:52:26.589443 *Tunnelblick: Expected disconnection occurred.

================================================================================

Down log:

13:52:25 *Tunnelblick:  **********************************************
13:52:25 *Tunnelblick:  Start of output from client.down.tunnelblick.sh
13:52:26 *Tunnelblick:  WARNING: Not restoring network settings because no saved Tunnelblick DNS information was found.
13:52:26 *Tunnelblick:  Flushed the DNS cache with dscacheutil -flushcache
13:52:26 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
13:52:26 *Tunnelblick:  End of output from client.down.tunnelblick.sh
13:52:26 *Tunnelblick:  **********************************************

================================================================================

Previous down log:

13:50:01 *Tunnelblick:  **********************************************
13:50:01 *Tunnelblick:  Start of output from client.down.tunnelblick.sh
13:50:01 *Tunnelblick:  Cancelled monitoring system configuration changes
13:50:01 *Tunnelblick:  Restored State:DNS
13:50:01 *Tunnelblick:  Restored Setup:DNS
13:50:01 *Tunnelblick:  Removed State:SMB
13:50:01 *Tunnelblick:  Restored DNS and SMB settings
13:50:01 *Tunnelblick:  Flushed the DNS cache with dscacheutil -flushcache
13:50:01 *Tunnelblick:  Notified mDNSResponder that the DNS cache was flushed
13:50:01 *Tunnelblick:  End of output from client.down.tunnelblick.sh
13:50:01 *Tunnelblick:  **********************************************

================================================================================

Network services:

An asterisk (*) denotes that a network service is disabled.
USB 10/100/1000 LAN
Wi-Fi
Bluetooth PAN
Thunderbolt Bridge

Wi-Fi Power (en0): On

================================================================================

ifconfig output:

lo0: flags=8049<UP,LOOPBACK,RUNNING,MULTICAST> mtu 16384
	options=1203<RXCSUM,TXCSUM,TXSTATUS,SW_TIMESTAMP>
	inet 127.0.0.1 netmask 0xff000000 
	inet6 ::1 prefixlen 128 
	inet6 fe80::1%lo0 prefixlen 64 scopeid 0x1 
	inet 10.254.254.254 netmask 0xff000000 
	nd6 options=201<PERFORMNUD,DAD>
gif0: flags=8010<POINTOPOINT,MULTICAST> mtu 1280
stf0: flags=0<> mtu 1280
XHC0: flags=0<> mtu 0
XHC20: flags=0<> mtu 0
en0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	ether f0:18:98:2e:66:36 
	inet 192.168.6.86 netmask 0xfffff800 broadcast 192.168.7.255
	media: autoselect
	status: active
p2p0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> mtu 2304
	ether 02:18:98:2e:66:36 
	media: autoselect
	status: inactive
awdl0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1484
	ether f6:95:28:e1:85:f7 
	inet6 fe80::f495:28ff:fee1:85f7%awdl0 prefixlen 64 scopeid 0x8 
	nd6 options=201<PERFORMNUD,DAD>
	media: autoselect
	status: active
en1: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=60<TSO4,TSO6>
	ether 2a:00:cd:80:90:01 
	media: autoselect <full-duplex>
	status: inactive
en2: flags=8963<UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	options=60<TSO4,TSO6>
	ether 2a:00:cd:80:90:00 
	media: autoselect <full-duplex>
	status: inactive
bridge0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether 2a:00:cd:80:90:01 
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x2
	member: en1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 9 priority 0 path cost 0
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 10 priority 0 path cost 0
	media: <unknown type>
	status: inactive
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 2000
	inet6 fe80::c721:9cde:8888:1397%utun0 prefixlen 64 scopeid 0xc 
	nd6 options=201<PERFORMNUD,DAD>
vboxnet0: flags=8943<UP,BROADCAST,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500
	ether 0a:00:27:00:00:00 
	inet 192.168.99.1 netmask 0xffffff00 broadcast 192.168.99.255

================================================================================

Non-Apple kexts that are loaded:

Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
  156    3 0xffffff7f965f4000 0xf0000    0xf0000    org.virtualbox.kext.VBoxDrv (6.0.10) FD628542-B180-34C4-AF29-CE70F4F9F587 <8 6 5 3 1>
  161    0 0xffffff7f9670d000 0x8000     0x8000     org.virtualbox.kext.VBoxUSB (6.0.10) 8626D518-71CF-3493-A7BC-5A5F24561525 <160 156 57 8 6 5 3 1>
  167    0 0xffffff7f967ee000 0x5000     0x5000     org.virtualbox.kext.VBoxNetFlt (6.0.10) E9EF0869-64B9-3C71-9133-4082F8227256 <156 8 6 5 3 1>
  171    0 0xffffff7f9680c000 0x6000     0x6000     org.virtualbox.kext.VBoxNetAdp (6.0.10) 9F140CBD-4713-3F03-AAD1-FBBC2B6C49A1 <156 6 5 1>

================================================================================

Quit Log:

2019-08-29 13:21:03.159945 applicationShouldTerminate: termination because of restart; delayed until 'shutdownTunnelblick' finishes)
2019-08-29 13:21:03.170888 shutDownTunnelblick: started.
2019-08-29 13:21:03.171186 shutDownTunnelblick: Starting cleanup.
2019-08-29 13:21:03.171432 cleanup: Entering cleanup
2019-08-29 13:21:03.171644 synchronized user defaults
2019-08-29 13:21:03.794585 Set up flag files for shutting down the computer and expecting all configurations to be disconnected
2019-08-29 13:21:03.795084 doDisconnectionsForShuttingDownComputer: Set 'expect disconnect 1 ALL'
2019-08-29 13:21:03.814631 Started disconnecting all configurations
2019-08-29 13:21:03.815151 Skipping cleanup because computer is shutting down or restarting
2019-08-29 13:21:03.815467 shutDownTunnelblick: Cleanup finished.
2019-08-29 13:21:03.815859 Finished shutting down Tunnelblick; allowing termination

================================================================================

Console Log:

2019-08-29 08:37:04.923053 Tunnelblick[75068] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 11:05:53.919112 Tunnelblick[75068] Preference 'Legacy Dev-skipWarningThatNotUsingSpecifiedOpenVPN' is TRUE, so not displaying the warning dialog.
2019-08-29 11:06:06.111499 Tunnelblick[75068] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 12:46:02.883416 Tunnelblick[75068] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-SLegacy Dev-Dtblk-SContents-SResources'
2019-08-29 12:46:18.615775 Tunnelblick[75068] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-SPROD-Dtblk-SContents-SResources'
2019-08-29 12:46:44.156573 Tunnelblick[75068] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 13:17:26.578402 Tunnelblick[75068] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 13:18:20.026662 Tunnelblick[75068] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-Sjbye--dev-Dtblk-SContents-SResources'
2019-08-29 13:18:35.271743 Tunnelblick[75068] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-SPROD-Dtblk-SContents-SResources'
2019-08-29 13:18:38.184108 Tunnelblick[75068] Preference 'Legacy Prod-skipWarningThatNotUsingSpecifiedOpenVPN' is TRUE, so not displaying the warning dialog.
2019-08-29 13:21:03.171376 Tunnelblick[75068] cleanup: Entering cleanup
2019-08-29 13:21:03.171608 Tunnelblick[75068] synchronized user defaults
2019-08-29 13:21:03.794494 Tunnelblick[75068] Set up flag files for shutting down the computer and expecting all configurations to be disconnected
2019-08-29 13:21:03.794998 Tunnelblick[75068] doDisconnectionsForShuttingDownComputer: Set 'expect disconnect 1 ALL'
2019-08-29 13:21:03.814528 Tunnelblick[75068] Started disconnecting all configurations
2019-08-29 13:21:03.814999 Tunnelblick[75068] Skipping cleanup because computer is shutting down or restarting
2019-08-29 13:21:03.815796 Tunnelblick[75068] Finished shutting down Tunnelblick; allowing termination
2019-08-29 13:24:39.802667 Tunnelblick[389] Tunnelblick: macOS 10.14.6; Tunnelblick 3.8.0 (build 5370)
2019-08-29 13:24:42.150942 Tunnelblick[389] Sparkle: ===== Tunnelblick =====
2019-08-29 13:24:42.151399 Tunnelblick[389] Sparkle: Verified appcast signature
2019-08-29 13:35:19.380835 Tunnelblick[389] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 13:43:37.183199 Tunnelblick[389] Preference 'skipWarningAboutDnsProblems' is TRUE, so not displaying the warning dialog.
2019-08-29 13:50:00.956674 Tunnelblick[389] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-Sjbye--dev-Dtblk-SContents-SResources'
2019-08-29 13:50:03.056173 Tunnelblick[389] Preference 'Legacy Dev-skipWarningThatNotUsingSpecifiedOpenVPN' is TRUE, so not displaying the warning dialog.
2019-08-29 13:52:25.014185 Tunnelblick[389] Set 'expect disconnect 1 -SLibrary-SApplication Support-STunnelblick-SUsers-Sjbye-SPROD-Dtblk-SContents-SResources'

```

**Comment by jkbullard** [comment]
Tunnelblick does not support having two configurations use "Set nameserver" simultaneously. And it warns you if you try to connect a second configuration with "Set nameserver" and you have a configuration that uses it already connected.

That is not something that will be changed in the foreseeable future, but I've marked this as an enhancement.

**Comment by jessebye** [comment]
@jkbullard thanks for clarifying. I didn't realize that "Set nameserver" wasn't supported for multiple simultaneous connections.

Doesn't this still qualify as a bug, since the user's internet connection is left in a bad state if they accidentally try to connect to multiple VPNs without realizing Tunnelblick doesn't support it?

**Comment by jkbullard** [comment]
@jessebye - Thanks. I just looked at the warning, and it doesn't say anything about the consequences of two or more connections using "Set nameserver", so I've removed the "Enhancement" tag..

However, I will probably fix the warning and then add the "Enhancement" tag back, since I will consider it more of an implementation restriction than a bug.

**Comment by jkbullard** [comment]
This was fixed in b9775963.
